src/views/Sale/components/SelfBidList/index.tsx

```diff
@@ -20,120 +20,161 @@ import { Table } from 'src/components/Table'
-            value:
-              numeral(formatBigInt(bid.amount, sale.tokenOut.decimals)).format('0.[0000]') + ' ' + sale.tokenIn.symbol,
-          })),
-        },
-      ]
-    : []
-
-  const isFairSaleHeadData = [
-    { title: 'Token Price' },
-    { title: 'Amount' },
-    { title: 'Total Tokens', flex: isMobile ? 5 : 3 },
-    { title: 'Status', flex: isMobile ? 1 : 3 },
-  ]
-
-  const isFairSaleOpenHeadData = [
-    { title: 'Token Price' },
-    { title: 'Amount' },
-    { title: 'Est. Invested' },
-    { title: sale.tokenOut.symbol },
-  ]
-  const isFairSaleBodyData = isFixed
-    ? []
-    : [
-        {
-          bids: (bids as FairSaleBid[]).map(bid => ({
-            bidPrice:
-              numeral(
-                formatBigInt(bid.tokenIn, sale.tokenIn.decimals) / formatBigInt(bid.tokenOut, sale.tokenOut.decimals)
-              ).format('0.[0000]') +
-              ' ' +
-              sale.tokenIn.symbol,
-            tokenIn: numeral(formatBigInt(bid.tokenIn, sale.tokenIn.decimals)).format('0.[0000]'),
-            tokenOut: numeral(formatBigInt(bid.tokenOut, sale.tokenOut.decimals)).format('0.[0.0000]'),
-
-            totalToken:
-              numeral(
-                formatBigInt(bid.tokenIn, sale.tokenIn.decimals) * formatBigInt(bid.tokenOut, sale.tokenOut.decimals)
-              ).format('0.[0000]') +
-              ' ' +
-              sale.tokenOut.symbol,
-          })),
-        },
-      ]
-
-  if (isFixed && isSaleClosed(sale)) {
-    return <Table headData={isFixedHeadData} bodyData={isFixedBodyData} isClosed={true} />
-  }
-
-  if (isFixed && isSaleOpen(sale)) {
-    return <Table headData={isFixedHeadDataOpen} bodyData={isFixedBodyDataOpen} />
-  }
-
-  if (isSaleOpen(sale) && !isFixed) {
-    return <Table headData={isFairSaleOpenHeadData} bodyData={isFairSaleBodyData as any} />
+  switch (sale.__typename) {
+    case 'FixedPriceSale': {
+      if (isSaleClosed(sale)) {
+        return (
+          <Table
+            headData={[
+              { title: 'Type', flex: isMobile ? 3.5 : 3 },
+              { title: 'Amount' },
+              { title: 'Value' },
+              { title: 'Status', flex: isMobile ? 2 : 3 },
+            ]}
+            bodyData={[
+              {
+                title: 'Withdraw',
+                purchases: {
+                  value:
+                    numeral(
+                      formatBigInt(
+                        (sale as GetFixedPriceSaleCommitmentsByUser_fixedPriceSaleCommitments_sale).tokenPrice,
+                        sale.tokenOut.decimals
+                      ) * formatBigInt((bids as FixedPriceSalePurchase).amount, sale.tokenOut.decimals)
+                    ).format('0.[0000]') +
+                    ' ' +
+                    sale.tokenOut.symbol,
+                  amount:
+                    numeral(formatBigInt((bids as FixedPriceSalePurchase).amount, sale.tokenOut.decimals)).format(
+                      '0.[0000]'
+                    ) +
```

Comment:
This value formatting could potentially be done by a helper function in future to keep this component cleaner, still this switch is a lot cleaner than before, good job